src/session.rs

```diff
@@ -342,11 +346,15 @@ pub struct UpdateRequest {
 }
 
 pub(crate) fn create_no_io_tsfn(
+    env: Env,
     js_function: JsFunction,
 ) -> Arc<Mutex<ThreadsafeFunction<(), ErrorStrategy::Fatal>>> {
-    Arc::new(Mutex::new(
-        js_function
+    Arc::new(Mutex::new({
+        let mut tsfn = js_function
             .create_threadsafe_function(0, |_ctx: ThreadSafeCallContext<()>| Ok(vec![()]))
-            .expect("Failed to create callback function"),
-    ))
+            .expect("Failed to create callback function");
+        // tell the runtime it can exit while this callback exists
+        tsfn.unref(&env).expect("Failed to unref callback function");
+        tsfn
+    }))
 }
```

Comment:
Can this be parameterized over the fn argument so that it can also be used for the update command?

If not, nbd - everyone likes a bit of pasta :spaghetti:

Reply:
Yeah, this was overdue for a refactor. Pushed helpers for building and calling all of these, will be useful for the connector PR as well.